openid-4-verifiable-presentations-1_0.md

```diff
@@ -595,11 +601,270 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+If the `values` property is present, the Wallet MUST return the claim only if the
+type and value of the claim both match for at least one of the elements in the array. 
+
+### Selecting Claims and Credentials {#vp_query_lang_processing_rules}
+
+The same basic logic applies for selecting claims and for selecting credentials,
+as detailed in the following. 
+
+Note: While this specification provides the mechanisms for requesting different
+sets of claims and credentials and request some of those optionally, it does not
+make assumptions about the user interface of the Wallet, for example, if users
+can select which claims to present or which combination of credentials to
+present.
+
+#### Selecting Claims {#selecting_claims}
+
+The following rules apply for selecting claims via `claims` and `claim_sets`:
+
+- If `claims` is not provided, the Verifier requests all claims existing
+  in the Credential.
+- If `claims` is provided, but `claim_sets` is not provided,
+  the Verifier requests all claims listed in `claims`.
+- Otherwise, the Verifier requests one combination of the claims listed in
+  `claim_sets`. Claims listed without a postfix are requested unconditionally.
+  Claims postfixed by `?` are requested optionally, i.e., it is up to the Wallet
+  to send the claim in the response or not. Claims postfixed by `?!` have to be
+  provided by the Wallet if they exist in the Credential.
+
+If the Wallet cannot deliver all non-optional claims requested by the Verifier
+according to these rules, it MUST NOT return the respective Credential.
+
+#### Selecting Credentials
+
+The following rules apply for selecting Credentials via `credentials` and `credential_sets`:
+
+- If `credential_sets` is not provided, the Verifier requests all
+  Credentials in `credentials` to be returned.
+- Otherwise, the Verifier requests one combination of the Credentials
+  listed in `credential_sets`, with optional credentials marked by the postfix `?`.
+
+Credentials not matching the respective constraints expressed within
+`credentials` MUST NOT be returned, i.e., they are treated as if
+they would not exist in the Wallet.
+
+If the Wallet cannot deliver all non-optional Credentials requested by the Verifier according to these rules, it MUST NOT
+return any credential(s).
+
+
+## Format-specific Properties {#format_specific_properties}
+
+The following format-specific properties are defined:
+
+### Format `vc+sd-jwt` {#format_vc_sd_jwt}
+
+`vct_values`:
+: OPTIONAL. An array of strings that specifies allowed values for
+the type of the requested Verifiable Credential. All elements in the array MUST
+be valid type identifiers as defined in [@!I-D.ietf-oauth-sd-jwt-vc]. The Wallet
+MAY return credentials that inherit from any of the specified types, following
+the inheritance logic defined in [@!I-D.ietf-oauth-sd-jwt-vc].
+
+### Format `mso_mdoc` {#format_mso_mdoc}
+
+`doctype_values`:
+: OPTIONAL. An array of strings that specifies allowed values for the
```

Comment:
This field should be mandatory as there is no default value for doctype.
It's also not clear what is meant with "valid doctype identifiers as defined in ISO 18013-5". The only mandatory requirement on the doctype identifier is that it's a text string. If that's what meant I think it's cleaner to be explicit about that.

Reply:
Should it be an array too? I would have thought it would only make sense for this to just be a string.

Reply:
Same question I have for SD-JWT i think both properties should support a singular value.

Reply:
Having a singular value makes sense to me as well.